base_tier_validation/tests/test_tier_validation.py

```diff
@@ -233,9 +243,63 @@ def test_12_approve_sequence(self):
         )
         # Request validation
         self.assertFalse(self.test_record.review_ids)
-        reviews = test_record.sudo(self.test_user_2.id).request_validation()
+        reviews = test_record.with_user(self.test_user_2.id).request_validation()
         self.assertTrue(reviews)
-        record = test_record.sudo(self.test_user_1.id)
+        record = test_record.with_user(self.test_user_1.id)
+        record.invalidate_cache()
         self.assertTrue(record.can_review)
         record.validate_tier()
         self.assertTrue(record.validated)
+
+    def test_13_onchange_review_type(self):
+        tier_def_id = self.tier_def_obj.create(
+            {
+                "model_id": self.tester_model.id,
+                "review_type": "individual",
+                "reviewer_id": self.test_user_1.id,
+                "definition_domain": "[('test_field', '>', 1.0)]",
+                "approve_sequence": True,
+            }
+        )
+        self.assertTrue(tier_def_id.reviewer_id)
+        tier_def_id.review_type = "group"
+        tier_def_id.onchange_review_type()
+        self.assertFalse(tier_def_id.reviewer_id)
+
+    def test_14_onchange_review_type(self):
+        tier_def_id = self.tier_def_obj.create(
+            {
+                "model_id": self.tester_model.id,
+                "review_type": "individual",
+                "reviewer_id": self.test_user_1.id,
+                "definition_domain": "[('test_field', '>', 1.0)]",
+                "approve_sequence": True,
+            }
+        )
+        self.assertTrue(tier_def_id.reviewer_id)
+        tier_def_id.review_type = "group"
+        tier_def_id.onchange_review_type()
+        self.assertFalse(tier_def_id.reviewer_id)
+
+    def test_15_review_user_count(self):
+        # Create new test record
+        test_record = self.test_model.create({"test_field": 2.5})
+        # Create tier definitions
+        self.tier_def_obj.create(
+            {
+                "model_id": self.tester_model.id,
+                "review_type": "individual",
+                "reviewer_id": self.test_user_1.id,
+                "definition_domain": "[('test_field', '>', 1.0)]",
+                "has_comment": True,
+            }
+        )
+        # Request validation
+        review = test_record.with_user(self.test_user_2).request_validation()
+        self.assertTrue(review)
+        self.assertTrue(self.test_user_1.get_reviews({"res_ids": review.ids}))
+        self.assertTrue(self.test_user_1.review_ids)
+        for review in self.test_user_1.review_ids:
+            review.invalidate_cache()
+        self.test_user_1.invalidate_cache()
+        self.test_user_1.review_user_count()
```

Comment:
no assert of the count after this?

Reply:
Absolutely! Indeed it's needed more tests to improve the code coverage.. I'm on it and soon will come a new commit with more tests and asserts